src/components/VCalendar/mixins/calendar-base.ts

```diff
@@ -0,0 +1,118 @@
+
+// Mixins
+import mixins from '../../../util/mixins'
+import Themeable from '../../../mixins/themeable'
+import Colorable from '../../../mixins/colorable'
+import Times from './times'
+import Mouse from './mouse'
+
+// Util
+import {
+  VTimestamp,
+  VTimestampFormatter,
+  validateTimestamp,
+  parseTimestamp,
+  getWeekdaySkips,
+  createDayList,
+  createNativeLocaleFormatter
+} from '../util/timestamp'
+
+/* @vue/component */
+export default mixins(Colorable, Themeable, Times, Mouse).extend({
+  name: 'calendar-base',
+
+  // depends on: maxDays
+
+  props: {
+    start: {
+      type: String,
+      required: true,
+      validate: validateTimestamp
+    },
+    end: {
+      type: String,
+      validate: validateTimestamp,
+      default: '0000-00-00'
+    },
+    weekdays: {
+      type: Array as () => number[],
+      default: () => [0, 1, 2, 3, 4, 5, 6]
+    },
+    hideHeader: {
+      type: Boolean,
+      default: false
+    },
+    shortWeekdays: {
+      type: Boolean,
+      default: true
+    },
+    weekdayFormat: {
+      type: Function, // VTimestampFormatter<string>,
+      default: null
+    },
+    dayFormat: {
+      type: Function, // VTimestampFormatter<string>,
+      default: null
+    },
+    locale: {
+      type: String,
+      default: 'en-us'
+    }
+  },
+
+  data: () => ({
+    defaultColor: 'primary'
+  }),
+
+  computed: {
+    weekdaySkips (): number[] {
+      return getWeekdaySkips(this.weekdays)
+    },
+    parsedStart (): VTimestamp {
+      return parseTimestamp(this.start) as VTimestamp
```

Comment:
You can remove `as VTimestamp` and just specify the same return type as `parseTimestamp`, `VTimestamp | null`

Reply:
That will have a cascading effect, having to change the signature of dozens of functions even though I know at this point it should be non-null - since I've validated its format this should always return a VTimestamp. The cast here was necessary due to linting.